client/my-sites/checkout/src/components/checkout-terms.tsx

```diff
@@ -56,18 +75,48 @@ export default function CheckoutTerms( { cart }: { cart: ResponseCart } ) {
 				isGiftPurchase={ Boolean( isGiftPurchase ) }
 				is100YearPlanPurchase={ has100YearPlan( cart ) }
 			/>
-			{ ! isGiftPurchase && <DomainRegistrationAgreement cart={ cart } /> }
-			{ ! isGiftPurchase && <DomainRegistrationHsts cart={ cart } /> }
-			{ ! isGiftPurchase && <DomainRegistrationDotGay cart={ cart } /> }
-			{ shouldShowRefundPolicy && <RefundPolicies cart={ cart } /> }
-			{ ! isGiftPurchase && <BundledDomainNotice cart={ cart } /> }
-			{ ! isGiftPurchase && <TitanTermsOfService cart={ cart } /> }
-			{ ! isGiftPurchase && <ThirdPartyPluginsTermsOfService cart={ cart } /> }
-			{ ! isGiftPurchase && <PlanTerms100Year cart={ cart } /> }
-			<EbanxTermsOfService />
-			{ shouldShowInternationalFeeNotice && <InternationalFeeNotice /> }
-			{ ! isGiftPurchase && <AdditionalTermsOfServiceInCart /> }
-			<JetpackSocialAdvancedPricingDisclaimer />
+			{ hasCheckoutVersion( '2' ) || showToSFoldableCard ? (
+				<>
+					{ ! isGiftPurchase && <AdditionalTermsOfServiceInCart /> }
+					{ ! isGiftPurchase && <DomainRegistrationAgreement cart={ cart } /> }
+					{ ! isGiftPurchase && <DomainRegistrationHsts cart={ cart } /> }
+					{ ! isGiftPurchase && <DomainRegistrationDotGay cart={ cart } /> }
+					<EbanxTermsOfService />
+					{ ! isGiftPurchase && <PlanTerms100Year cart={ cart } /> }
+					{ ! isGiftPurchase && <ThirdPartyPluginsTermsOfService cart={ cart } /> }
+					{ ! isGiftPurchase && <TitanTermsOfService cart={ cart } /> }
+
+					<TermsCollapsedContent>
+						<FoldableCard
+							clickableHeader={ true }
+							compact
+							className="checkout__terms-foldable-card"
+							header={ translate( 'Read more' ) }
+							screenReaderText={ translate( 'Read more' ) }
+						>
+							{ shouldShowRefundPolicy && <RefundPolicies cart={ cart } /> }
+							{ ! isGiftPurchase && <BundledDomainNotice cart={ cart } /> }
+							<InternationalFeeNotice />
```

Comment:
We decided earlier (https://github.com/Automattic/wp-calypso/pull/84031/files#r1396328926) that we'd remove the `shouldShowInternationalFeeNotice` condition so that there would at least be one thing always rendered inside the card.

However, I just noticed that `InternationalFeeNotice` can itself decide to be hidden, which means there's still a case where the foldable card will be empty. 😞 

To solve this I think we're going to have to refactor each of these components so that their visibility is controlled entirely from outside the component, then we can use the logic that determines that visibility to decide if the card will be empty or not, and therefore when to hide it. (Alas, there's no effective way to check if a component has rendered something.) That's not easy, since both `BundledDomainNotice` and `RefundPolicies` have a lot of possible conditions and `JetpackSocialAdvancedPricingDisclaimer` we already saw was complex.

Let's chat about this on Monday and see if we can come up with a plan so that this doesn't delay the test.